### DISCUSSION TITLE
Vulnerability Issue

### DISCUSSION THREAD

**Discussion by satya-devloper** [strimzi/strimzi-kafka-operator#7787] [Q&A]
## 👋  Welcome to Strimzi Discussions!

We want to use _Discussions_ instead of the _Question_ type issues. The discussions provide much better format for questions than the issues. They for example:
* Make it easier for other users with similar problems to chime in
* Unlike issues they do not need to be closed
* Do not mix with bugs and enhancements and make it easier for us to track and fix those

You could of course also use our [#strimzi Slack channel](https://slack.cncf.io/) or out [user mailing list](https://lists.cncf.io/g/cncf-strimzi-users/topics).

Please also make sure to follow our [Code of Conduct](https://github.com/strimzi/strimzi-kafka-operator/blob/master/CODE_OF_CONDUCT.md).

_Originally posted by @scholzj in https://github.com/strimzi/strimzi-kafka-operator/discussions/4099_

**Comment by satya-devloper** [comment]
Hi @scholz , I m using this strimzi  kafka image (quay.io/strimzi/kafka:0.32.0- kafka -3.2.0) in my project .But I m getting lots of vulnerabilities. Like 6 unaddressed vulnerability in golang, 7 vulnerability in rhel and many more with with 1 or 2 vulnerability. Plz suggest which image should I use so that I can get less vulnerability

**Reply by scholzj** [reply]
You can wait for the next release or build the images from sources.

**Reply by satya-devloper** [reply]
Are there any web link which explicitly tell which version of other application like go, CentOS, openssl , curl.., the particular image is using.
Also if i build the image from source also do I need to download all the projects under strimzi , as I can see that CentOS 7 is being used in event bridge. So probably there is dependency between kafka-operator and event-bridge, if I am correct .

**Reply by scholzj** [reply]
We do not use CentOS 7 for anything as far as I know, we are not written in Golang, and there is nothing what Strimzi has what is called `event-bridge`.

**Reply by satya-devloper** [reply]
My bad, its strimzi-kafka-bridge . When I downloaded the other image like (quay.io/strimzi/kafka:0.29.0- kafka -3.2.0). What i can see in the image is go version of 1.17.1 , it was used. But when i upgraded image to latest version , I can see go version of 1.19. 
In that sense i was asking the above question

**Reply by scholzj** [reply]
Go version of what?

**Reply by satya-devloper** [reply]
Also ,If i see the dockerfile of strimzi-kafka-bridge, it is trying to pull the base image of Centos:7, and the 7 defects which are coming are coming from centos only. Thats why I m thinking does strimzi-kafka-operator somehow linked with strimzi-kafka bridge. If linked does it mean we need to download all the repos inside the strimzi, in order to build it from source

**Reply by satya-devloper** [reply]
> Go version of what?

Means somehow (quay.io/strimzi/kafka:0.29.0- kafka -3.2.0) this image is internally using golang version of 1.17.1. But after upgrade it is picking the version 1.19 , but still 1.19 version of golang has CVE defects

**Reply by scholzj** [reply]
As you can see, the bridge is not using CentOS 7: https://github.com/strimzi/strimzi-kafka-bridge/blob/3e08e474ee224715ba7fb28aa9deb945a376ebfd/Dockerfile#L1 ... as for the golang, you really need to be more precise what exactly are you talking about.

**Reply by satya-devloper** [reply]
Thanks a lot, somehow I was looking at https://github.com/strimzi/strimzi-kafka-bridge/tree/81e8c76c2293d4b83b63059bff4b9371064b03bf, this unluckily which is very older. 
Can you plz provide some steps in order to build the image from source. So that I can replace the required version of application so that defect should not come in  our project

**Reply by scholzj** [reply]
Please read the developer docs to learn how to build the project.

**Reply by satya-devloper** [reply]
Sure will check, Thanks for your reply.
